### DISCUSSION TITLE
Yet another, quite enhanced rtl2mqtt script

### DISCUSSION THREAD

**Discussion by sheilbronn** [merbanan/rtl_433#2440] [Show and tell]
Being a longtime and passionate user of rtl_433 to pick up sensor data from my home and neighbourhood, I'd like to give back and to offer the daemon/script that I built around the rtl_433 output: https://github.com/sheilbronn/rtl2mqtt

From the README:

* Suppression of repeated (duplicate) messages, e.g. from motion sensors, within a short time-frame.
* Support for Home Assistant MQTT auto-discovery announcements for new sensors
* Temperature output is transformed to SI units (=Celsius) and rounded for less flicker.
* Streamlined/removed mostly unnecessary content in the original JSON messages, e.g. no time stamp or checksum code.
* Frequent unchanged MQTT messages from temperature or humidity sensors within a certain time  frame are suppressed.
* MQTT topic contains - configurably - the sensor's id.
* Configurable upload of weather sensor data to [Weather Underground (WU)](https://www.wunderground.com)
* Enhanced logging and debugging into a device-specific subdirectory structure, easing later source device analysis. 
* A MQTT state and a log channel for the bridge is provided giving regular statistics and on certain events of the bridge itself.
* Command line options to be used in every invocation can be put into `~/.config/rtl2mqtt`

I've been using it successfuly with my OpenHab installation for a long time.  
The script is quite big and makes heavy use of bash features as far as possible for performance (few subprocess invocations).  
Please feel free to comment and whether you'd like me to add information to the wiki.  
And many thanks for rtl_433!

**Comment by zuckschwerdt** [comment]
This sounds great! The filtering and transformation are much requested features and we always answered "someone need to write a bridge script" :) Thanks!
Add a link to the Wiki please. https://github.com/merbanan/rtl_433/wiki/Built-with-rtl_433%21

**Reply by sheilbronn** [reply]
Thanks. I added a line to the wiki.

**Comment by rct** [comment]
@sheilbronn -- Thanks for sharing this. I look forward to testing it.

I'm amazed you were able to do all of this in Bash!
